Two maybe-uninitialized warnings in vfs.cc

Using gcc 8.3.0 I'm seeing two uninitialized warnings in vfc.cc

```
/home/tiledb/TileDB/tiledb/sm/filesystem/vfs.cc: In member function 'tiledb::sm::Status tiledb::sm::VFS::filelock_unlock(const tiledb::sm::URI&) const':
/home/tiledb/TileDB/tiledb/sm/filesystem/vfs.cc:421:16: error: 'should_unlock' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   if (!st.ok() || !should_unlock) {
       ~~~~~~~~~^~~~~~~~~~~~~~~~~
/home/tiledb/TileDB/tiledb/sm/filesystem/vfs.cc:429:37: error: 'fd' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     return posix_.filelock_unlock(fd);
                                     ^
[ 36%] Building CXX object tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/filter/compression_filter.cc.o
cc1plus: all warnings being treated as errors
make[5]: *** [tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/build.make:401: tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/filesystem/vfs.cc.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[4]: *** [CMakeFiles/Makefile2:125: tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/all] Error 2
make[3]: *** [Makefile:141: all] Error 2
make[2]: *** [CMakeFiles/tiledb.dir/build.make:119: tiledb-prefix/src/tiledb-stamp/tiledb-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:176: CMakeFiles/tiledb.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

--- a/tiledb/sm/filesystem/vfs.cc
+++ b/tiledb/sm/filesystem/vfs.cc
@@ -415,8 +415,8 @@ Status VFS::filelock_unlock(const URI& uri) const {
   std::unique_lock<std::mutex> lck(filelock_mtx_);
 
   // Decrement the lock counter and return if the counter is still > 0.
-  bool should_unlock;
-  filelock_t fd;
+  bool should_unlock = false;
+  filelock_t fd = INVALID_FILELOCK;
   Status st = decr_lock_count(uri, &should_unlock, &fd);
   if (!st.ok() || !should_unlock) {
     return st;

